scripts/rebuild.sh

```diff
@@ -3,19 +3,32 @@
 set -e
 
 cd -P -- "$(dirname -- "${BASH_SOURCE[0]}")"
-if [ ! -e ../vcpkg ]
+repo_url=https://github.com/Microsoft/vcpkg
+vcpkg_commit=$(git ls-remote $repo_url | head -n1 | awk '{ print $1 }')
+filename="commit.txt"
+if [ ! -f "$filename" ]
+then
+    # create file with default value
+    echo "0" > "$filename"
+    echo "File created with default value: $filename"
+fi
+
+content=$(cat commit.txt)
+
+if [ $vcpkg_commit != $content ]
 then
-    vcpkg_commit=d10d511f25620ca0f315cd83dcef6485efc63010
     git init ../vcpkg
     git -C ../vcpkg fetch --depth 1 https://github.com/Microsoft/vcpkg $vcpkg_commit
     git -C ../vcpkg checkout FETCH_HEAD
+    echo $vcpkg_commit > commit.txt
 fi
 
+githubtoken=$github_token
 npm ci
 rm -rf ../en
 node generatePages.js
 node generateDocs.js
 node validateLinks.js
-node generateGitHubStars.js ../vcpkg $1
+node generateGitHubStars.js ../vcpkg $1 $githubtoken
```

Comment:
First: this adds more dynamic behavior -- the stars will change on every run. Updating the stars with a github token should be done as a separate pipeline step, only in the update pipeline. For developers, they should only be using the cached stars information.
Second: `$1` was intended to be the PAT / token input.
```js
console.log('Usage: node generateGitHubStars.js <vcpkg-root> [GITHUB_PAT]');
```

Reply:
thanks, I will update the stars information in a separate pipeline step.